src/pocketmine/Player.php

```diff
@@ -3123,8 +3061,21 @@ public function handleDataPacket(DataPacket $packet){
 				$this->setDataFlag(self::DATA_FLAGS, self::DATA_FLAG_ACTION, false); //TODO: check if this should be true
 
 				switch($packet->event){
-					case 9: //Eating
-						$this->consumeHeldItem();
+					case EntityEventPacket::USE_ITEM: //Eating
+						$slot = $this->inventory->getItemInHand();
+
+						if($slot->canBeConsumed()){
```

Comment:
This check is unnecessary. It is unreasonable that this happens, and if this really happens, just let the plugin handle it (default cancelled, as on line 3070)